openjdk hotspot build system std.h fix

Backport from upstream from jdk9

    OpenJDK / jdk9 / hs-rt / hotspot
    
    changeset 6671:465eb6bc50d1
    
    8046818: Hotspot build system looking for sdt.h in the wrong place
    Reviewed-by: sla, dholmes
    author	mikael
    date	Thu, 03 Jul 2014 09:16:56 -0700 (14 months ago)
    parents	35de71d2af77
    children	da4514678bcf
    files	make/linux/makefiles/dtrace.make
    diffstat	1 files changed, 8 insertions(+), 1 deletions(-) [+]

Upstream-Status: Backport
---
 openjdk/hotspot/make/linux/makefiles/dtrace.make |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

--- openjdk/hotspot/make/linux/makefiles/dtrace.make
+++ openjdk/hotspot/make/linux/makefiles/dtrace.make
@@ -40,7 +40,14 @@
 ifneq ($(ALT_SDT_H),)
   SDT_H_FILE = $(ALT_SDT_H)
 else
-  SDT_H_FILE = /usr/include/sys/sdt.h
+  ifeq ($(USE_CLANG), true)
+    # Clang doesn't support the -print-sysroot option and there is no known equivalent
+    # option, so fall back to using / as sysroot
+    SDT_SYSROOT=
+  else
+    SDT_SYSROOT=$(shell $(CXX) -print-sysroot)
+  endif
+  SDT_H_FILE = $(SDT_SYSROOT)/usr/include/sys/sdt.h
 endif
 DTRACE_ENABLED = $(shell test -f $(SDT_H_FILE) && echo $(SDT_H_FILE))
 REASON = "$(SDT_H_FILE) not found"
